packages/transaction-controller/src/helpers/IncomingTransactionHelper.ts

```diff
@@ -289,7 +289,7 @@ export class IncomingTransactionHelper {
 
   #getBlockNumberKey(additionalKeys: string[]): string {
     const currentChainId = this.#getCurrentChainId();
-    const currentAccount = this.#getCurrentAccount().toLowerCase();
+    const currentAccount = this.#getCurrentAccount()?.toLowerCase();
```

Comment:
I appreciate this is the quick fix but it's definitely just hiding the issue since we should be able to rely on this callback to always return a value as per the typing.